linker/shared/src/main/scala/org/scalajs/linker/backend/wasmemitter/CoreWasmLib.scala

```diff
@@ -616,6 +626,119 @@ object CoreWasmLib {
     newFunctionBuilder(functionID, OriginalName(functionID.toString()))
   }
 
+  private def genBoxBoolean()(implicit ctx: WasmContext): Unit = {
```

Comment:
Why don't we box boolean using `i31ref` too?

Reply:
No, we can't. That would make the values appear to JS as `0` and `1`, instead of `false` and `true`.